common/src/main/java/org/apache/drill/common/types/Types.java

```diff
@@ -806,16 +806,20 @@ public static boolean isSortable(MinorType type) {
     return typeBuilder;
   }
 
+  public static boolean isSameType(MajorType type1, MajorType type2) {
```

Comment:
Yes I got the idea, and this suggestion is part of my personal beliefs. I like to hide everything until it's required to be used elsewhere. Since it's minor comment, I'm ok with leaving it as is.